script.js

```diff
@@ -0,0 +1,80 @@
+// CHALLENGE 1
+// `// given an array of post objects create a function that returns a new array of posts, each object in the new array should have keys renamed as per the example. Original array of posts should not be mutated.`
+
+// `// @param {{id: number, createdBy: number, commentNo: number}[]}`
+
+// `// @returns {{postId: number, postCreatedBy: number }[]}`
+
+const postsArray = [
+  {
+    id: 123,
+    createdBy: "user16",
+    commentNo: 5,
+  },
+  {
+    id: 456,
+    createdBy: "user20",
+    commentNo: 5,
+  },
+  {
+    id: 789,
+    createdBy: "user32",
+    commentNo: 5,
+  },
+];
+
+const newPostArr = (arr) => {
+  [...arr].map((item) => {
+    return {
+      postsId: item.id,
+      postsCreated: item.createdBy,
+    };
+  });
+};
+
+// Challenges using the SWAPI API: https://swapi.dev/
+
+// CHALLENGE 2
+// Fetch all people from the API, create a function that takes in the results array and returns a new array only with people with height above 166
+
+const fetchData = async (searchTerm) => {
+  const response = await fetch(`https://swapi.dev/api/${searchTerm}/`);
+
+  const data = await response.json();
+  return data.results;
+};
+
+const heightFilter = async () => {
+  const people = await fetchData("people");
+  let newArr = [];
+
+  people.map((item) => {
```

Comment:
this looks like it would work but there are a few unnecessary steps happening:

1. Iterators .map and .filter return new arrays, so there is no need to create an empty array and push to it from inside the iterator callback function
2. .map might not be the best choice here, since it returns an array of the same length as the original one, instead of having map with an if.check we can use `.filter`, like this:
```js
const newArr = people.filter((item) => {
      return item.height > 166
})
```

`.filter` takes in a predicate function - a function that return a boolean, and creates a new array with only elements for which that predicate function evaluated to true